### PR TITLE
Add control of recursion desired flag for DNS probes

### PR DESCRIPTION
**Pull request by dgl**
This defaults recursion desired to on to preserve existing behaviour,
but allows it to be configured.

Asked for in a comment in #474 and generally useful.

Signed-off-by: David Leadbeater <dgl@dgl.cx>

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -189,6 +189,8 @@ query_name: <string>
 [ query_type: <string> | default = "ANY" ]
 [ query_class: <string> | default = "IN" ]
 
+[ recursion_desired: <boolean> | default = true ]
```
Comment (SuperQ) [inline CONFIGURATION.md:192]: Let's simplify the name to just "recursion".

```suggestion
[ recursion: <boolean> | default = true ]
```

File: config/config.go [hunk 2]
```diff
@@ -264,8 +265,9 @@ type DNSProbe struct {
 	TransportProtocol  string           `yaml:"transport_protocol,omitempty"`
 	QueryClass         string           `yaml:"query_class,omitempty"` // Defaults to IN.
 	QueryName          string           `yaml:"query_name,omitempty"`
-	QueryType          string           `yaml:"query_type,omitempty"`   // Defaults to ANY.
-	ValidRcodes        []string         `yaml:"valid_rcodes,omitempty"` // Defaults to NOERROR.
+	QueryType          string           `yaml:"query_type,omitempty"`        // Defaults to ANY.
+	RecursionDesired   bool             `yaml:"recursion_desired,omitempty"` // Defaults to true.
```
Comment (SuperQ) [inline config/config.go:269]: Same here, this can be just `Recursion`.

**Review by SuperQ — CHANGES_REQUESTED** [review]
Minor naming nit, otherwise LGTM.

File: prober/utils_test.go [hunk 3]
```diff
@@ -162,7 +162,7 @@ func TestChooseProtocol(t *testing.T) {
 	registry = prometheus.NewPedanticRegistry()
 
 	ip, _, err = chooseProtocol(ctx, "ip4", false, "ipv6.google.com", registry, logger)
-	if err != nil && err.Error() != "address ipv6.google.com: no suitable address found" {
+	if err != nil && err.Error() != "address ipv6.google.com: no suitable address found" && err.Error() != "lookup ipv6.google.com: no such host" {
```
Comment (electron0zero) [inline prober/utils_test.go:165]: I don't understand this? can you please explain why we need this check?

Comment (dgl) [reply prober/utils_test.go:165]: Was a fix for go 1.17 as the error message changed, #864 covers it too, so will rebase it out.

Comment (SuperQ) [reply prober/utils_test.go]: I actually found a better fix.

**Review by SuperQ — APPROVED** [review]
LGTM

**Pull request merged**